plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroHiveFileUtils.java

```diff
@@ -280,6 +283,14 @@ private static List<Schema> removeDuplicateNullSchemas(List<Schema> childSchemas
         return prunedSchemas;
     }
 
+    static Map<String, String> getLowerCaseToGivenFieldName(Schema schema)
+    {
+        //Lower case top level fields to allow for manually set avro schema (passed in via avro_schema_literal or avro_schema_url) to have uppercase field names
+        ImmutableMap.Builder<String, String> lowerCaseToGivenNameBuilder = ImmutableMap.builder();
+        schema.getFields().stream().map(Schema.Field::name).forEach(fieldName -> lowerCaseToGivenNameBuilder.put(fieldName.toLowerCase(Locale.ENGLISH), fieldName));
+        return lowerCaseToGivenNameBuilder.buildOrThrow(); // this will throw if two fields nave the same name in lowercase
```

Comment:
Don't see anything in the spec disallowing it. What's the best way to throw a better error from build or throw? I usually just see it used as a kinda `verify`. And this is a situation where I struggle to see how a user could write data that we could query here without running into issues?

To get here the user must:
1) Set up schema with duplicate canonical names
2) Init table in unsupported way on Trino to avoid throwing an error for duplicate canonical names
3) Not use Hive to write any data?

Maybe if they use like some kinda of FlinkHiveWriter or something?